util/progress/progressui/init.go

```diff
@@ -0,0 +1,46 @@
+package progressui
+
+import (
+	"os"
+	"runtime"
+
+	"github.com/morikuni/aec"
+)
+
+var noColorIsSet bool
+var buildkitColorIsSet bool
+var colorRun aec.ANSI
+var colorCancel aec.ANSI
+var colorWarning aec.ANSI
+var colorError aec.ANSI
+var envColorString string
+
+func init() {
+	// As recommended on https://no-color.org/
+	_, noColorIsSet = os.LookupEnv("NO_COLOR")
+	// Loosely based on the standard set by Linux LS_COLORS.
+	_, buildkitColorIsSet = os.LookupEnv("BUILDKIT_COLORS")
+
+	if noColorIsSet {
+		colorRun = termColorMap["default"]
```

Comment:
Is this right or should `NO_COLOR` skip the aec color functions completely?

Reply:
Probably. I actually mention this in a comment above. It probably does need to skip `aec` completely, but I haven't dug into how to accomplish that exactly, yet. Unfortunately, `aec` does not directly support this at the moment. I'll take a look as soon as I can.

Also, happy to get a general direction/recommendation from anyone who is familiar with the codebase.

Reply:
Thanks @tonistiigi . I'll see if I can't get this working when I have a bit of extra time this week.

Reply:
I think I found a simple `NO_COLOR` implementation. This should be good for another round of reviews.